typewise_alert.py

```diff
@@ -1,46 +1,60 @@
+cooling_type_limits = {
+    "HI_ACTIVE_COOLING": {'lower_limit': 0, 'upper_limit': 45},
+    "MED_ACTIVE_COOLING": {'lower_limit': 0, 'upper_limit': 40},
+    "PASSIVE_COOLING": {'lower_limit': 0, 'upper_limit': 35},
+}
 
-def infer_breach(value, lowerLimit, upperLimit):
-  if value < lowerLimit:
-    return 'TOO_LOW'
-  if value > upperLimit:
-    return 'TOO_HIGH'
-  return 'NORMAL'
-
-
-def classify_temperature_breach(coolingType, temperatureInC):
-  lowerLimit = 0
-  upperLimit = 0
-  if coolingType == 'PASSIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 35
-  elif coolingType == 'HI_ACTIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 45
-  elif coolingType == 'MED_ACTIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 40
-  return infer_breach(temperatureInC, lowerLimit, upperLimit)
-
-
-def check_and_alert(alertTarget, batteryChar, temperatureInC):
-  breachType =\
-    classify_temperature_breach(batteryChar['coolingType'], temperatureInC)
-  if alertTarget == 'TO_CONTROLLER':
-    send_to_controller(breachType)
-  elif alertTarget == 'TO_EMAIL':
-    send_to_email(breachType)
-
-
-def send_to_controller(breachType):
-  header = 0xfeed
-  print(f'{header}, {breachType}')
-
-
-def send_to_email(breachType):
-  recepient = "a.b@c.com"
-  if breachType == 'TOO_LOW':
-    print(f'To: {recepient}')
-    print('Hi, the temperature is too low')
-  elif breachType == 'TOO_HIGH':
-    print(f'To: {recepient}')
-    print('Hi, the temperature is too high')
+email_alert_message = {"TOO_LOW": {'receiver_address': 'recepient_address@organisation.com',
+                                   'alert_message': "Temperature is too low. Requesting immediate action"},
+                       "TOO_HIGH": {'receiver_address': 'recepient_address@organisation.com',
+                                    'alert_message': "Temperature is too high. Requesting immediate action"}
+                       }
+
+
+def check_and_alert(alert_target, battery_char, temperature_in_Celsius):
+    breach_type = \
+        classify_temperature_breach(battery_char['coolingType'], temperature_in_Celsius)
+    if breach_type != "NORMAL":
+        return target_alert[alert_target](breach_type)
+
+
+def classify_temperature_breach(cooling_type, temperature_in_celsius):
+    if check_if_input_is_valid(cooling_type, temperature_in_celsius):
+        cooling_temp_range = cooling_type_limits[cooling_type]
+        return infer_breach(temperature_in_celsius, cooling_temp_range['lower_limit'],
+                            cooling_temp_range['upper_limit'])
+    else:
+        return 'INVALID_INPUT'
```

Comment:
You have coded the invalid-input case but not tested for it. Does it mean it's not important what it returns?